### ISSUE TITLE
packer azure-arm crash - unexpected EOF

### ISSUE THREAD

**Issue by segonzalez-dev** [hashicorp/packer#7988]
**Packer Version:**
```bash
Packer v1.4.2
```

**Host platform:**
Bug found on these two platforms:

* Ubuntu 18.04.3 LTS
* MacOS Mojave

**Debug log output from crash.log (Sensitive values had been replaced/removed):**
https://gist.github.com/segonzalez-dev/7ee784218e2197cfda615c847eade41c

**Sample packer script extract:**
```json
{
  "type": "azure-arm",
  "communicator": "ssh",
  "ssh_username": "{{user `ssh_username`}}",
  "ssh_password": "{{user `ssh_password`}}",
  "pause_before_connecting": "2m",
  
  "client_id": "{{user `arm_client_id`}}",
  "client_secret": "{{user `arm_client_secret`}}",
  "subscription_id": "{{user `arm_subscription_id`}}",
  "tenant_id": "{{user `arm_tenant_id`}}",
  
  "managed_image_resource_group_name": "{{user `arm_resource_group`}}",
  "managed_image_name": "{{user `image_name`}}",
  
  "os_type": "Linux",
  "image_publisher": "Canonical",
  "image_offer": "UbuntuServer",
  "image_sku": "18.04-LTS",
  
  "location": "{{user `arm_region`}}",
  "vm_size": "Standard_B1ms"
}
```

**Error message:**
```bash
2019/08/14 13:18:14 packer: panic: interface conversion: interface {} is nil, not string
2019/08/14 13:18:14 packer:
2019/08/14 13:18:14 packer: goroutine 117 [running]:
2019/08/14 13:18:14 packer: github.com/hashicorp/packer/builder/azure/arm.(*Builder).Run(0xc0001c6e70, 0x3c5b160, 0xc000345840, 0x3c732a0, 0xc00042cbc0, 0x3bfaf00, 0xc000440320, 0x0, 0x0, 0x0, ...)
2019/08/14 13:18:14 packer: 	/Users/mmarsh/Projects/packer/builder/azure/arm/builder.go:305 +0x4384
2019/08/14 13:18:14 packer: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc0003ac990, 0x1, 0xc0000521e0, 0x0, 0x0)
2019/08/14 13:18:14 packer: 	/Users/mmarsh/Projects/packer/packer/rpc/builder.go:109 +0x231
2019/08/14 13:18:14 packer: reflect.Value.call(0xc000094e40, 0xc00000e178, 0x13, 0x36aa961, 0x4, 0xc0000b0f18, 0x3, 0x3, 0xc000344f40, 0x3, ...)
2019/08/14 13:18:14 packer: 	/usr/local/go/src/reflect/value.go:447 +0x461
2019/08/14 13:18:14 packer: reflect.Value.Call(0xc000094e40, 0xc00000e178, 0x13, 0xc000274718, 0x3, 0x3, 0xc0004bf5c0, 0xc000274730, 0x404e35)
2019/08/14 13:18:14 packer: 	/usr/local/go/src/reflect/value.go:308 +0xa4
2019/08/14 13:18:14 packer: net/rpc.(*service).call(0xc000090440, 0xc0000a80f0, 0xc000052520, 0xc000052530, 0xc000100700, 0xc0003583a0, 0x2c84720, 0xc0000521bc, 0x18a, 0x2be05c0, ...)
2019/08/14 13:18:14 packer: 	/usr/local/go/src/net/rpc/server.go:384 +0x14e
2019/08/14 13:18:14 packer: created by net/rpc.(*Server).ServeCodec
2019/08/14 13:18:14 packer: 	/usr/local/go/src/net/rpc/server.go:481 +0x42b
2019/08/14 13:18:14 /home/user/bin/packer: plugin process exited
2019/08/14 13:18:14 [INFO] (telemetry) ending azure-arm
2019/08/14 13:18:14 ui error: [1;31mBuild 'azure-arm' errored: unexpected EOF[0m
2019/08/14 13:18:14 machine readable: error-count []string{"1"}
2019/08/14 13:18:14 ui error:
==> Some builds didn't complete successfully and had errors:
2019/08/14 13:18:14 machine readable: azure-arm,error []string{"unexpected EOF"}
2019/08/14 13:18:14 ui error: --> azure-arm: unexpected EOF
2019/08/14 13:18:14 ui:
==> Builds finished but no artifacts were created.
2019/08/14 13:18:14 Cancelling builder after context cancellation context canceled
2019/08/14 13:18:14 Error cancelling builder: connection is shut down
```

**Comment by SwampDragons** [comment]
I'm 99% sure this is fixed in the 1.4.3 release I just cut. Can you verify for me?

**Comment by segonzalez-dev** [comment]
Hello @SwampDragons,

I was able to create an Azure image with version 1.4.3. Please mark this issue as closed.

Thanks for your help
